clang/lib/DPCT/AnalysisInfo.cpp

```diff
@@ -2366,45 +2366,36 @@ inline void DeviceFunctionDeclInModule::insertWrapper() {
   {
     auto FunctionBlock = Printer.block();
     Printer.indent();
-#ifdef _WIN32
-    Printer << "__declspec(dllexport) ";
-#endif
-    Printer << "void " << FuncName << "_wrapper(" << MapNames::getClNamespace()
+    Printer << "DPCT_EXPORT void " << FuncName << "_wrapper(" << MapNames::getClNamespace()
             << "queue &queue, const " << MapNames::getClNamespace()
             << "nd_range<3> &nr, unsigned int localMemSize, void "
                "**kernelParams, void **extra)";
     if (HasBody) {
-      Printer << "{";
+      auto for_each_parameter = [&](auto F) {
+	auto it = getParametersInfo().begin();
+	for (int i = 0;
+	     it != getParametersInfo().end();
+	     ++it, ++i) {
+	  F(i, it->second);
+	}
+      };
+
+      Printer << " {";
       {
         auto BodyBlock = Printer.block();
         Printer.newLine();
-        for (auto It = getParametersInfo().begin();
-             It != getParametersInfo().end(); It++) {
-          Printer.line(It->first + " " + It->second + ";");
-        }
-        Printer.line("if(kernelParams){");
-        {
-          auto IfBlock = Printer.block();
-          auto Counter = 0;
-          for (auto It = getParametersInfo().begin();
-               It != getParametersInfo().end(); It++) {
-            Printer.line(It->second + " = (" + It->first + ")kernelParams[" +
-                         std::to_string(Counter) + "];");
-            Counter += 1;
-          }
-        }
-        Printer.line("} else {");
-        {
-          auto ElseBlock = Printer.block();
-          std::string ExtraOffsetStr = "sizeof(void*)";
-          for (auto It = getParametersInfo().begin();
-               It != getParametersInfo().end(); It++) {
-            Printer.line(It->second + " = (" + It->first + ")(extra + " +
-                         ExtraOffsetStr + ");");
-            ExtraOffsetStr += " + sizeof(" + It->first + ")";
-          }
-        }
-        Printer.line("}");
+	requestFeature(HelperFeatureEnum::Util_args_selector, FilePath);
```

Comment:
Should this requestFeature line be moved before line 2369 that generates DPCT_EXPORT?  Perhaps Util_args_selector should be renamed to Util_kernel_wrapper, since it is not just for argument selection.